web/src/utils/formatting.ts

```diff
@@ -206,7 +207,21 @@ const formatDateTick = (
   }
 
   switch (timeRange) {
-    case TimeRange.H72:
+    case TimeRange.H72: {
+      const [localHours, localMinutes] = getLocalTime(date, timezone);
+      if (localHours === 0 && localMinutes === 0) {
+        // Display date name when midnight
+        return new Intl.DateTimeFormat(lang, {
+          month: 'short',
+          day: 'numeric',
+          timeZone: 'UTC',
+        }).format(date);
+      }
+      return new Intl.DateTimeFormat(lang, {
+        timeStyle: 'short',
+        timeZone: timezone,
+      }).format(date);
+    }
```

Comment:
Mind if I refractor this whole function in a follow up PR so we are only creating one Intl object (and less code)?

Reply:
go for it!